### PR TITLE
Change and cleaning eslint rules to be more compatible with "airbnb-eslint-conf".

### PR DESCRIPTION
**Pull request by swistach**
### Context
After our migration from traceur to babel, we had to change the eslint configuration. Unfortunately, some of the important rules we marked as "notice" instead of "error". After a few months after our migration, this is a perfect time to adjust our ruleset.

### How has this been tested?
Clone repository and go inside directory.
Run `npm run eslint`.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Related issue(s):
1. #4536 

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "scripts": {
     "clean": "rimraf commonjs es coverage",
     "lint": "eslint --rule \"no-restricted-globals: 0\" test && eslint --rule \"no-restricted-globals: 0\" --ext .unit.js --ext .e2e.js src/ && eslint --ignore-pattern \"src/plugins/**/test/**\" src/",
```
Comment (wszymanski) [inline package.json]: Could be changed according to comment:  https://github.com/handsontable/handsontable/pull/4724#pullrequestreview-89749164

File: .eslintrc [hunk 2]
```diff
@@ -10,65 +10,22 @@
...
   "rules": {
     "class-methods-use-this": 0,
-    "comma-dangle": 0,
-    "consistent-return": 1,
-    "dot-notation": 2,
-    "eol-last": 2,
-    "eqeqeq": 1,
     "func-names": 0,
     "import/extensions": 0,
     "import/no-extraneous-dependencies": 0,
```
Comment (wszymanski) [inline .eslintrc]: Lines 19 and 20 could be changed according to comment: #4724 (review)

File: package.json (hunk 1 above)
Comment (budnix) [reply package.json]: This line was introduced while fixing linter for Windows. After changing this line it should be tested if everything works.

File: test/e2e/Core_update.spec.js [hunk 3]
```diff
@@ -42,18 +38,14 @@ describe('Core_updateSettings', () => {
...
       },
-      cells(row, col, prop) {
-        if (row === 0 && col === 0) {
-          return {
-            type: 'numeric'
-          };
-        }
-      }
+      cells(row, col) {
+        return row === 0 && col === 0 ? { type: 'numeric' } : {};
```
Comment (wszymanski) [inline test/e2e/Core_update.spec.js]: Should it return `{}` in the second case?

File: test/e2e/Core_validate.spec.js [hunk 4]
```diff
@@ -941,26 +930,15 @@ describe('Core_validate', () => {
...
         callback(validationResult);
-
-        /* Setting this variable has to be async, because we are not interested in when the validation happens, but when
-         the callback is being called. Since internally all the callbacks are processed asynchronously (even if they are
-         synchronous) end of validator function is not the equivalent of whole validation routine end.
-         If it still sounds weird, take a look at HandsontableTextEditorClass.prototype.finishEditing method.
-         */
-
-        setTimeout(() => {
-          validated = true;
```
Comment (wszymanski) [inline test/e2e/Core_validate.spec.js]: Can it be removed?

**Comment by wszymanski** [comment]
There is a lot of useful changes which should help us with code maintenance 👍 . Some of the changes can be done now, but some need discussion.

**Comment by swistach** [comment]
I think there are too many conflicts to continue improving here. The better idea is creating a few smaller issues and sending changes one by one in small pieces.

**Pull request closed without merging**